### PR TITLE
feat: add kafka consumer to Pokeshop

### PR DESCRIPTION
**Pull request by danielbdias**
This PR adds a worker that consumes a Kafka stream and executes the ImportPokemon use case.

To run the trace-based tests on this API, you can run the command:
```sh
make run/tracetests
```

Loom: https://www.loom.com/share/296a89f7ce324a2fa64026f427c114ab?sid=211e1850-6514-4001-a036-414435d5f490

## Changes

- Added streamWorker to consume Kafka messages
- Updated node version to latest LTS (the current one will reach EOL next month)
- Added Trace-based tests to validate API functionalities

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:16.14.0-alpine as build-ui
+FROM node:20.5.1-alpine as build-ui
```
Comment (mathnogueira) [inline Dockerfile:1]: I can only imagine how many things stopped working :boom:

File: api/src/services/stream.service.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    for (let key in kafkaHeaders) {
+      let value = kafkaHeaders[key] || "";
+      if (value.toString !== undefined) {
+        value = value.toString();
+      }
+      headers[key] = value;
+    }
+
+    return headers
+  }
```
Comment (mathnogueira) [inline api/src/services/stream.service.ts:52]: I was expecting your usual wasted hours counter here :laughing: 

> This block of code owns me X hours of my life
> \- Daniel Dias

**Review by mathnogueira — APPROVED** [review]
Great work!

File: api/src/services/stream.service.ts (hunk 2 above)
Comment (danielbdias) [reply api/src/services/stream.service.ts:52]: Indeed 😄 !

This block of code owns me 2 days of my life!

**Pull request merged**